src/handlers/add.ts

```diff
@@ -13,9 +13,14 @@ export namespace AddHandler {
 
     const id = nanoid();
     const allTasks: Array<Task> = await getTasks();
-    const by = args.by ? new Date(args.by).toISOString() : undefined;
 
-    allTasks.push({ id: id, done: false, description: args.description, by: by });
+    try {
+      const by = args.by ? new Date(args.by).toISOString() : undefined;
+    } catch (e: unknown) {
+      console.error("Oops! Please type valid date");
+      process.exit(1);
+    }
+    allTasks.push({ id: id, done: false, description: args.description, by: args.by });
```

Comment:
This doesn't seem to be using the date you parsed in the above code. Instead it uses what the user directly provided.

Reply:
Same as above

Reply:
Yeah...I misunderstood about this part and now I could fix it! Thank you and please check it